applications/ShapeOptimizationApplication/custom_utilities/vertex_morphing_mapper.h

```diff
@@ -221,68 +223,62 @@ class VertexMorphingMapper
         {
+            // Extract sub-model part for damping
+            ModelPart& damping_sub_mdpa = extractModelPart(damping_regions[region_itr][0]);
+
             // Read settings for current edge
-            std::string edge_sub_model_part_name = extractString(damped_edges[edge_itr][0]);
-            KRATOS_WATCH(edge_sub_model_part_name);
-            bool damp_in_X = damped_edges[edge_itr][1];
-            bool damp_in_Y = damped_edges[edge_itr][2];
-            bool damp_in_Z = damped_edges[edge_itr][3];
-            std::string damping_function_type = extractString(damped_edges[edge_itr][4]);
-            double damping_radius = extractDouble(damped_edges[edge_itr][5]);
-
-            // Check if sub-model part collecting all edge nodes exist
-            if(!mr_opt_model_part.HasSubModelPart(edge_sub_model_part_name))
-                KRATOS_THROW_ERROR(std::invalid_argument, "Specified sub-model part for edge damping not found: ",edge_sub_model_part_name);
+            bool damp_in_X = damping_regions[region_itr][1];
+            bool damp_in_Y = damping_regions[region_itr][2];
+            bool damp_in_Z = damping_regions[region_itr][3];
+            std::string damping_function_type = extractString(damping_regions[region_itr][4]);
+            double damping_radius = extractDouble(damping_regions[region_itr][5]);
 
             // Prepare damping function
             DampingFunction damping_function( damping_function_type, damping_radius );
 
-            // Loop over all edge nodes specified in a dedicated sub-model part 
-            for ( ModelPart::NodeIterator node_itr = mr_opt_model_part.GetSubModelPart(edge_sub_model_part_name).NodesBegin(); 
-                  node_itr != mr_opt_model_part.GetSubModelPart(edge_sub_model_part_name).NodesEnd(); 
-                  ++node_itr )
+            // Loop over all nodes in specified damping sub-model part 
+            for ( ModelPart::NodeIterator node_itr = damping_sub_mdpa.NodesBegin(); node_itr != damping_sub_mdpa.NodesEnd(); ++node_itr )
             {
+                // Get node information
+                int i_ID = node_itr->Id();
+                ModelPart::NodeType& damping_node_i = mr_opt_model_part.Nodes()[i_ID];
+                array_3d i_coord = damping_node_i.Coordinates();
 
+                // Perform spatial search for current node
+                unsigned int number_of_nodes_affected;
+                number_of_nodes_affected = nodes_tree.SearchInRadius( damping_node_i, 
+                                                                        damping_radius, 
+                                                                        nodes_affected.begin(),
+                                                                        resulting_squared_distances.begin(), 
+                                                                        max_nodes_affected );    
+
```

Comment:
you constructed the `nodes_tree` with `max_nodes_affected` as bucket size. In this case, I think the `SearchInRadius` has the same complexity as a for loop. I suggest using a smaller bucket size to take advantage of the kdtree search structure.